spooq/transformer/annotator.py

```diff
@@ -0,0 +1,247 @@
+        - Apply comments from ``columns_mapping`` within the :py:class:`~spooq.transformer.mapper.Mapper` transformer
+        - Apply comments from existing silver table to gold dataframe
+          within the :py:class:`~spooq.transformer.mapper.Mapper` transformer
+        - Combine any of them
+
+    Raises
+    ------
+        ColumnNotFound: If column is missing from the dataframe and missing_column_handling is set to ``raise_error``
+    """
+
+    def __init__(
+        self,
+        comments_mapping: Dict[str, str] = None,
+        mode: AnnotatorMode = AnnotatorMode.upsert,
+        missing_column_handling: MissingColumnHandling = MissingColumnHandling.raise_error,
+        sql_source_table_identifier: str = None,
+    ):
+        super().__init__()
+        self.comments_mapping = comments_mapping or {}
+        self.mode = mode
+        self.missing_column_handling = missing_column_handling
+        self.sql_source_table_identifier = sql_source_table_identifier
+
+    def transform(self, input_df: DataFrame) -> DataFrame:
+        if self.sql_source_table_identifier:
+            self.logger.info(
+                f"sql_source_table_identifier was provided: "
+                f"Trying to load existing comments from source table at <{self.sql_source_table_identifier}> ..."
+            )
+            existing_comments_mapping = load_comments_from_metastore_table(
+                sql_source_table_identifier=self.sql_source_table_identifier
+            )
+            self.logger.info(
+                f"Trying to add {len(existing_comments_mapping)} existing comments "
+                f"from {self.sql_source_table_identifier} to provided comments_mapping!"
+            )
+            for column, comment in existing_comments_mapping.items():
+                if not column in input_df.columns:
+                    self.logger.debug(
+                        f"Skipping to apply comment ('{comment}') fetched from the sql_source_table because "
+                        f"column: '{column}' was not found in the dataframe."
+                    )
+                    continue
+
+                if column in self.comments_mapping:
+                    self.logger.debug(
+                        f"Comment for column: '{column}' is both defined in the sql_source_table ('{comment}') "
+                        f"and the mapping ('{self.comments_mapping[column]}')."
+                    )
+                    if self.mode == AnnotatorMode.upsert:
+                        self.logger.debug("Using comment from mapping.")
+                        continue
+
+                self.logger.debug("Using comment from sql_source_table.")
+                self.comments_mapping[column] = comment
+
+            self.logger.info(
+                f"Added {len(set(input_df.columns).intersection(set(existing_comments_mapping)))} "
+                f"comments from {self.sql_source_table_identifier}!"
+            )
```

Comment:
```suggestion
            for column, existing_comment in existing_comments_mapping.items():
                if not column in input_df.columns:
                    self.logger.debug(
                        f"Skipping to apply comment ('{existing_comment}') fetched from the sql_source_table because "
                        f"column: '{column}' was not found in the dataframe."
                    )
                    continue

                if column in self.comments_mapping:
                    self.logger.debug(
                        f"Comment for column: '{column}' is both defined in the sql_source_table ('{existing_comment}') "
                        f"and the mapping ('{self.comments_mapping[column]}')."
                    )
                    if self.mode == AnnotatorMode.upsert:
                        self.logger.debug("Using comment from mapping.")
                        continue

                self.logger.debug("Using comment from sql_source_table.")
                self.comments_mapping[column] = existing_comment

            self.logger.info(
                f"Added {len(set(input_df.columns).intersection(set(existing_comments_mapping)))} "
                f"comments from {self.sql_source_table_identifier}!"
            )
```